File: ios/Runner/Model/Contract/SongModelContract.swift
//
//  SongModelContract.swift
//  Runner
//
//  Created by t k on 2024/08/12.
//

import MediaPlayer

protocol SongModelContract {
    var media: MPMediaItemCollection { get }

    var id: SongId { get }

    var name: String { get }

    var albumId: AlbumId { get }

    var albumName: String { get }

    var artistId: ArtistId { get }

    var artistName: String { get }

    var imageId: String { get }

    var artwork: MPMediaItemArtwork? { get }

    var duration: Double { get }
    
    var trackNumber: Int { get }
}

File: android/app/src/main/kotlin/dev/tcode/thinmpf/model/contract/SongModelContract.kt
package dev.tcode.thinmpf.model.contract

import android.net.Uri
import dev.tcode.thinmpf.model.valueObject.AlbumId
import dev.tcode.thinmpf.model.valueObject.ArtistId
import dev.tcode.thinmpf.model.valueObject.SongId

interface SongModelContract {
    val id: SongId
    val name: String
    val albumId: AlbumId
    val albumName: String
    val artistId: ArtistId
    val artistName: String
    val imageId: String
    val imageUri: Uri
    val mediaUri: Uri
    val duration: Int
    val trackNumber: Double
}